Source/ACE.Server/Factories/LootGenerationFactory.cs

```diff
@@ -1373,96 +1373,101 @@ private static double GetManaCMod(int tier)
-                        missileMod = .01;
-                    else if (chance > 80)
-                        missileMod = .005;
-                    else
-                        missileMod = 0;
-                    break;
-                case 5:
-                    chance = ThreadSafeRandom.Next(0, 1000);
-                    if (chance > 950)
-                        missileMod = .01;
-                    else if (chance > 800)
-                        missileMod = .005;
-                    else
-                        missileMod = 0;
-                    break;
-                case 6:
-                    chance = ThreadSafeRandom.Next(0, 1000);
-                    if (chance > 975)
-                        missileMod = .020;
-                    else if (chance > 900)
-                        missileMod = .015;
-                    else if (chance > 800)
-                        missileMod = .010;
-                    else if (chance > 700)
-                        missileMod = .005;
-                    else
-                        missileMod = 0;
-                    break;
-                case 7:
-                    chance = ThreadSafeRandom.Next(0, 1000);
-                    if (chance > 990)
-                        missileMod = .030;
-                    else if (chance > 985)
-                        missileMod = .025;
-                    else if (chance > 950)
-                        missileMod = .020;
-                    else if (chance > 900)
-                        missileMod = .015;
-                    else if (chance > 850)
-                        missileMod = .01;
-                    else if (chance > 800)
-                        missileMod = .005;
-                    else
-                        missileMod = 0;
-                    break;
-                default: // tier 8
-                    chance = ThreadSafeRandom.Next(0, 1000);
-                    if (chance > 998)
-                        missileMod = .04;
-                    else if (chance > 994)
-                        missileMod = .035;
-                    else if (chance > 990)
-                        missileMod = .03;
-                    else if (chance > 985)
-                        missileMod = .025;
-                    else if (chance > 950)
-                        missileMod = .02;
-                    else if (chance > 900)
-                        missileMod = .015;
-                    else if (chance > 850)
-                        missileMod = .01;
-                    else if (chance > 800)
-                        missileMod = .005;
-                    else
-                        missileMod = 0;
-                    break;
+                switch (tier)
+                {
+                    case 1:
+                    case 2:
+                        magicMissileDefenseMod = 0;
+                        break;
+                    case 3:
+                        int chance = ThreadSafeRandom.Next(0, 100);
```

Comment:
technically this is rolling between 0-100 here, inclusive, meaning 101 possible values

should probably be 1-100

Reply:
Fixed and pushed